Documentation/ceph-debugging.md

```diff
@@ -0,0 +1,231 @@
+```
+
+#### Misc <!-- omit in toc -->
+Logs can be followed and watched in real-time which is useful debugging a problem you are trying to
+reproduce as shown in the example below.
+```
+kubectl --namespace rook-ceph logs <pod-name> --follow
+```
+
+You can view logs for previously-running containers in a pod. This is useful when a container/Pod
+has encountered a failure and restarted where the currently running container has yet to fail. The
+failure from the previous container can still be viewed.
+```
+kubectl --namespace rook-ceph logs <pod-name> --previous
+```
+
+### Describe Deployments, Pods, and other Kubernetes resources
+All resources including Pods in Kubernetes can be "described". This will give more detailed
+information about a resource. This information is critical to understanding how the Rook Operator
+has configured Ceph resources, particularly Deployments.
+
+#### Basic describe <!-- omit in toc -->
+Basically detailed information about resources is easy to obtain.
+```sh
+kubectl --namespace rook-ceph describe <resource-type> <resource-name>
+# e.g., kubectl --namespace rook-ceph describe deployment rook-ceph-operator
+# e.g., kubectl --namespace rook-ceph describe pod rook-ceph-mon-a-69cd744894-zc9jm
+```
+
+#### JSON description (more advanced info) <!-- omit in toc -->
+If you wish to use get JSON output from a detailed description (to use jq on the results or for
+other reasons), that is possible. Additionally there are some much more advanced and not often
+useful fields which are left out of "describe' output that are shown in JSON output.
+```sh
+kubectl --namespace rook-ceph get --output=json <resource-type> <resource-name>
+```
+
+### Stop the operator
+
+Stopping the operator is done usually when the Rook-Ceph cluster will be modified manually in some
+way. Do this when you don't want the operator to attempt to repair the cluster, either because it
+would revert manual steps or because the operator could get into an error state because of changes
+to the cluster. Stopping the operator is done in practice by scaling down the number of deployment
+replicas to zero, thus keeping the original deployment intact but stopping the pod from continuing
+to execute.
+
+```sh
+kubectl --namespace rook-ceph scale deployment --replicas=0 rook-ceph-operator
+```
+
+This should be undone by scaling the deployment back to 1 (and only 1) replica. This will start the
+operator pod and thereby start the Rook operator running again.
+```sh
+kubectl --namespace rook-ceph scale deployment --replicas=1 rook-ceph-operator
+```
+
+### Restart the operator
+In order to do a simple restart of the Rook operator, it does not need scaled down and scaled back
+up again. This is as simple as deleting the Operator pod. The operator deployment will restart the
+deleted pod almost immediately, thus restarting the Operator.
+```sh
+kubectl --namespace rook-ceph delete pod --selector app=rook-ceph-operator
+```
+
+### Restart a daemon pod
```

Comment:
Restarting a pod is mentioned in specific troubleshooting sections, not sure we need its own section.